pycheribuild/projects/cross/apache.py

```diff
@@ -0,0 +1,144 @@
+                                              ac_cv_struct_rlimit="yes",
+                                              ac_cv_func_sem_open="yes",
+                                              apr_cv_process_shared_works="yes",
+                                              apr_cv_mutex_robust_shared="yes",
+                                              # XXX: This might be yes on Linux
+                                              apr_cv_tcp_nodelay_with_cork="no",
+                                              )
+
+        self.run_cmd("./buildconf", cwd=self.source_dir)
+        super().configure(**kwargs)
+
+class BuildApache(CrossCompileAutotoolsProject):
+    target = "apache"
+    project_name = "apache"
+
+    repository = GitRepository("https://github.com/apache/httpd.git",
+                               default_branch="2.4.x")
+
+    dependencies = ["apr", "pcre"]
+
+    native_install_dir = DefaultInstallDir.IN_BUILD_DIRECTORY
+    cross_install_dir = DefaultInstallDir.ROOTFS_LOCALBASE
+
+    def configure(self, **kwargs):
+        self.configure_args.extend([
+            "--enable-layout=FreeBSD",
+            "--enable-http",
+            "--enable-mod-ssl",
+            "--with-expat=" + str(self.install_dir),
+            "--with-pcre=" + str(BuildPcre.get_instance(self).install_dir),
+            "--with-apr=" + str(BuildApr.get_instance(self).install_dir),
+            ])
+
+        # The cross-compile always assumes this is true (so one
+        # can't normally cross-compile Apache), and it's a fatal
+        # error, so just lie always.
+        self.configure_environment.update(ap_cv_void_ptr_lt_long="no")
+
+        # mod_ssl doesn't use -L from --with-expat
+        self.LDFLAGS.append("-L" + str(self.install_dir / "lib"))
+
+        if not self.compiling_for_host() and False:
+            # Can't determine these when cross-compiling
+            self.configure_environment.update(ac_cv_file__dev_zero="yes",
+                                              ac_cv_mmap__dev_zero="yes",
+                                              # XXX: This might be yes on Linux
+                                              ac_cv_func_setpgrp_void="no",
+                                              ac_cv_struct_rlimit="yes",
+                                              ac_cv_func_sem_open="yes",
+                                              apr_cv_process_shared_works="yes",
+                                              apr_cv_mutex_robust_shared="yes",
+                                              # XXX: This might be yes on Linux
+                                              apr_cv_tcp_nodelay_with_cork="no",
+                                              )
+
+        self.run_cmd("./buildconf", cwd=self.source_dir)
+        super().configure(**kwargs)
+
+        # gen_test_char rules in server/ assume a native build
+        if not self.compiling_for_host():
+            self.run_cmd(str(self.host_CC), "-DCROSS_COMPILE", "-c",
```

Comment:
CFLAGS too? Both for things like optimisation, but also because host_CC may be a clang that doesn't target the host by default.

Reply:
I think we generally assume that `host_CC` can build files for the current system without any target/abi/etc. flags.

Reply:
Hmm, do we have a suitable host CFLAGS?

Reply:
I don't think so. I guess this is just a simple build tool? In that case shouldn't it be fine to just use the host compiler defaults?

Reply:
It just uses <ctype.h> to print out tables of 'isprint()', etc.  Technically using the host version of the tables probably isn't correct, but is probably sufficient for cheribuild.